integration-tests/smoke/ccip_batching_test.go

```diff
@@ -0,0 +1,295 @@
+	require.NoError(t, ccdeploy.AddLaneWithDefaultPrices(e.Env, state, sourceChain2, destChain))
+
+	const (
+		numMessages = 5
+	)
+	var (
+		startSeqNum map[uint64]ccipocr3.SeqNum = map[uint64]ccipocr3.SeqNum{
+			sourceChain1: 1,
+			sourceChain2: 1,
+		}
+		endSeqNum map[uint64]ccipocr3.SeqNum = map[uint64]ccipocr3.SeqNum{
+			sourceChain1: ccipocr3.SeqNum(numMessages),
+			sourceChain2: ccipocr3.SeqNum(numMessages),
+		}
+	)
+
+	t.Run("batch data only messages from single source", func(t *testing.T) {
+		err := sendMessages(
+			ctx,
+			t,
+			e.Env.Chains[sourceChain1],
+			e.Env.Chains[sourceChain1].DeployerKey,
+			state.Chains[sourceChain1].OnRamp,
+			state.Chains[sourceChain1].Router,
+			state.Chains[sourceChain1].Multicall3,
+			destChain,
+			numMessages,
+			common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+		)
+		require.NoError(t, err)
+
+		_, err = ccdeploy.ConfirmCommitWithExpectedSeqNumRange(
+			t,
+			e.Env.Chains[sourceChain1],
+			e.Env.Chains[destChain],
+			state.Chains[destChain].OffRamp,
+			nil,
+			ccipocr3.NewSeqNumRange(startSeqNum[sourceChain1], endSeqNum[sourceChain1]),
+		)
+		require.NoErrorf(t, err, "failed to confirm commit from chain %d", sourceChain1)
+
+		startSeqNum[sourceChain1] = endSeqNum[sourceChain1] + 1
+		endSeqNum[sourceChain1] = startSeqNum[sourceChain1] + ccipocr3.SeqNum(numMessages) - 1
+	})
+
+	t.Run("batch data only messages from multiple sources", func(t *testing.T) {
+		var (
+			wg   sync.WaitGroup
+			mx   sync.Mutex
+			errs []error
+		)
+
+		wg.Add(1)
+		go func(sourceChainSelector uint64) {
+			defer wg.Done()
+			err := sendMessages(
+				ctx,
+				t,
+				e.Env.Chains[sourceChainSelector],
+				e.Env.Chains[sourceChainSelector].DeployerKey,
+				state.Chains[sourceChainSelector].OnRamp,
+				state.Chains[sourceChainSelector].Router,
+				state.Chains[sourceChainSelector].Multicall3,
+				destChain,
+				numMessages,
+				common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+			)
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}(sourceChain1)
+
+		wg.Add(1)
+		go func(sourceChainSelector uint64) {
+			defer wg.Done()
+			err := sendMessages(
+				ctx,
+				t,
+				e.Env.Chains[sourceChainSelector],
+				e.Env.Chains[sourceChainSelector].DeployerKey,
+				state.Chains[sourceChainSelector].OnRamp,
+				state.Chains[sourceChainSelector].Router,
+				state.Chains[sourceChainSelector].Multicall3,
+				destChain,
+				numMessages,
+				common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+			)
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}(sourceChain2)
+
+		wg.Wait()
+
+		for _, err := range errs {
+			require.NoError(t, err)
+		}
+
+		// confirm the commit reports
+		var (
+			sourceChain1Report *offramp.OffRampCommitReportAccepted
+			sourceChain2Report *offramp.OffRampCommitReportAccepted
+		)
+		errs = nil
+
+		wg.Add(1)
+		go func() {
```

Comment:
Similarly to the previous one, this involves a lot of duplication. I guess it might be very useful for other tests, so might be useful to extract that to some global function that accepts multiple sources and waits for them in async manner